R/run.R

```diff
@@ -301,15 +301,19 @@ run_one <- function(bm,
       read_only = read_only,
       run_id = run_id,
       run_name = run_name,
+      batch_id = batch_id,
       run_reason = run_reason
     ),
     keep.null = TRUE
   )
   result <- do.call(run_script, run_script_args)
-  # If JSON reading fails because `run_bm()` errored, `batch_id` will not be populated
-  if (is.null(result$batch_id)) {
-    result$batch_id <- batch_id
+
+  if (!identical(Sys.getenv("TESTTHAT"), "true") && !is.null(result$error)) {
+    cat("Result errored; quitting with status = 1")
+    cat(result$error)
+    quit(status = 1L)
   }
```

Comment:
Previously `run_script()` would exit with status = 1, but `run_one()` (which is what labs/benchmarks calls) does not. This needs to get picked up by labs/benchmarks and escalated; hopefully that will make it through the conbench CLI.

Reply:
Reverting; this causes more problems than it fixes. Having nice exit statuses is important for CLIs, but not for libraries where we have more sophisticated error handling already.